server/src/main/java/org/apache/calcite/avatica/server/HttpServer.java

```diff
@@ -659,6 +686,12 @@ public Builder withRemoteUserExtractor(RemoteUserExtractor remoteUserExtractor)
       return withAuthentication(AuthenticationType.DIGEST, properties, allowedRoles);
     }
 
+    public Builder<T> withCustomAuthentication(AvaticaServerConfiguration config) {
```

Comment:
I think Karan was just following what I had elsewhere in here (`withDigestAuthentication(String, String[])`, `withBasicAuthentication(String, String[])`, and the inconsistently named `withSpnego(String, String, String[])`).

Sounds like a great follow-on to me :)

Reply:
Unsure here about the comment. What do you think looks neater?

Reply:
IMO, what you have now is fine. I think what Andrew suggested would be a nice improvement, but we would have to make sure we don't break compatibility among other things. That would just be good to separate out and handle later.

Reply:
Great. I feel that `withServerConfiguration()` method name is not a very good representative of what it does internally and hence would like to prefer my current approach only. I can probably overload the private method `withAuthentication()` so that we should not cause any trouble, but I think its good for now.